src/Xamarin.Android.Build.Tasks/Linker/MonoDroid.Tuner/LinkDesignerBase.cs

```diff
@@ -117,6 +126,52 @@ protected void FixType (TypeDefinition type, TypeDefinition localDesigner)
 			}
 		}
 
+		protected void RemoveFieldsFromType (TypeDefinition type, ModuleDefinition module)
+		{
+			List<FieldDefinition> fields = new List<FieldDefinition> ();
+			TypeReference intArray = new ArrayType (module.TypeSystem.Int32);
+			foreach (var field in type.Fields) {
+				if (field.FieldType.IsArray) {
+					continue;
+				}
+				fields.Add (field);
+			}
+			foreach (var field in fields) {
+				LogMessage ($"Removing {type.Name}::{field.Name}");
+				type.Fields.Remove (field);
+			}
```

Comment:
I think this also would work, and you wouldn't need an intermediate list:
```suggestion
			for (int i = fields.Length - 1; i >= 0; i--) {
				var field = fields [i];
				if (field.FieldType.IsArray) {
					continue;
				}
				LogMessage ($"Removing {type.Name}::{field.Name}");
				type.Fields.RemoveAt (i);
			}
```